src/zenml/zen_stores/secrets_stores/azure_secrets_store.py

```diff
@@ -0,0 +1,633 @@
+            KeyError: If the secret does not exist.
+            RuntimeError: If the Azure Key Vault API returns an unexpected
+                error.
+        """
+        azure_secret_id = self._get_azure_secret_id(secret_id)
+
+        try:
+            azure_secret = self.client.get_secret(
+                azure_secret_id,
+            )
+        except ResourceNotFoundError:
+            raise KeyError(f"Secret with ID {secret_id} not found")
+        except HttpResponseError as e:
+            raise RuntimeError(
+                f"Error fetching secret with ID {secret_id} {e}"
+            )
+
+        # The _convert_azure_secret method raises a KeyError if the
+        # secret is tied to a workspace or user that no longer exists. Here we
+        # simply pass the exception up the stack, as if the secret was not found
+        # in the first place, knowing that it will be cascade-deleted soon.
+        assert azure_secret.properties.tags is not None
+        return self._convert_azure_secret(
+            tags=azure_secret.properties.tags,
+            values=azure_secret.value,
+        )
+
+    def list_secrets(
+        self, secret_filter_model: SecretFilterModel
+    ) -> Page[SecretResponseModel]:
+        """List all secrets matching the given filter criteria.
+
+        Note that returned secrets do not include any secret values. To fetch
+        the secret values, use `get_secret`.
+
+        Args:
+            secret_filter_model: All filter parameters including pagination
+                params.
+
+        Returns:
+            A list of all secrets matching the filter criteria, with pagination
+            information and sorted according to the filter criteria. The
+            returned secrets do not include any secret values, only metadata. To
+            fetch the secret values, use `get_secret` individually with each
+            secret.
+
+        Raises:
+            ValueError: If the filter contains an out-of-bounds page number.
+            RuntimeError: If the Azure Key Vault API returns an unexpected
+                error.
+        """
+        # The Azure Key Vault API does not natively support any of the
+        # filtering, sorting or pagination options that ZenML supports. The
+        # implementation of this method therefore has to fetch all secrets from
+        # the Key Vault, then apply the filtering, sorting and pagination on
+        # the client side.
+
+        # The metadata will always contain at least the filter criteria
+        # required to exclude everything but Azure secrets that belong to the
+        # current ZenML deployment.
+        results: List[SecretResponseModel] = []
+
+        try:
+            all_secrets = self.client.list_properties_of_secrets()
+            for secret_property in all_secrets:
+                try:
+                    # NOTE: we do not include the secret values in the
+                    # response. We would need a separate API call to fetch
```

Comment:
It seems like this returns values for some implementations and not for others? This is hella confusing IMO, that should be consistent.

Reply:
No, absoluely not. `list_secrets` never returns any secret values. This is not only documented in the API and Client methods and implemented consistently the same across all back-ends, it is also tested in the integration tests, and we ran and passed all tests on all of the back-ends.